Source/resizeablerectitem.cpp

```diff
@@ -109,7 +109,32 @@ void ResizeableRectItem::paint( QPainter* aPainter, const QStyleOptionGraphicsIt
     }
 
     // Draw the rect outline.
-    aPainter->drawRect( cRect );
+    aPainter->drawRect(cRect);
+
+    // Draw the object name on the rect if no image is provided
+    if (m_Pixmap.isNull())
+    {
+        const auto objectName = mMapObject->mObjectStructureType.c_str();
+        for (;;)
+        {
+            QFont f = aPainter->font();
+            QFontMetricsF fm(f);
+            QRectF textRect = fm.boundingRect(cRect, Qt::AlignCenter | Qt::TextWrapAnywhere, objectName);
+
+            auto size = f.pointSize();
+            if (textRect.width() > cRect.width() ||
+                textRect.height() > cRect.height())
+            {
+                f.setPointSize(--size);
```

Comment:
need to handle the case of size <= 0 when the text can't fit else this will inf loop

Reply:
Or set some min size to give up at, I'd assume at some point it get so small it doesn't matter if it fits or not